v3/src/components/graph/components/scatter-plot-utils.ts

```diff
@@ -0,0 +1,83 @@
+  const { dataset: data, yAttributeIDs: yAttrIDs = [], hasY2Attribute, numberOfPlots = 1 } = dataConfiguration || {}
+  const numExtraPrimaryBands = dataConfiguration?.numRepetitionsForPlace('bottom') ?? 1
+  const numExtraSecondaryBands = dataConfiguration?.numRepetitionsForPlace('left') ?? 1
+  const topSplitID = dataConfiguration?.attributeID("topSplit") ?? ""
+  const rightSplitID = dataConfiguration?.attributeID("rightSplit") ?? ""
+  const xAttrID = dataConfiguration?.attributeID("x") ?? ""
+  const xScale = layout.getAxisScale("bottom") as ScaleLinear<number, number>
+  const y1Scale = layout.getAxisScale("left") as ScaleLinear<number, number>
+  const y2Scale = hasY2Attribute ? layout.getAxisScale("rightNumeric") as ScaleLinear<number, number> : undefined
+  const rightScale = layout.getAxisScale('rightCat') as ScaleBand<string> | undefined
+  const topScale = layout.getAxisScale('top') as ScaleBand<string> | undefined
+
+  function getXCoord(caseID: string) {
+    const xValue = data?.getNumeric(caseID, xAttrID) ?? NaN
+    const topValue = data?.getStrValue(caseID, topSplitID) ?? ''
+    const topCoord = (topValue && topScale?.(topValue)) || 0
+    return xScale(xValue) / numExtraPrimaryBands + topCoord
+  }
+
+  function getYCoord(caseID: string, plotNum = 0) {
+    const yAttrID = yAttrIDs[plotNum]
+    const yValue = data?.getNumeric(caseID, yAttrID) ?? NaN
+    const yScale = y2Scale && plotNum === numberOfPlots - 1 ? y2Scale : y1Scale
+    const rightValue = data?.getStrValue(caseID, rightSplitID) ?? ''
+    const rightCoord = ((rightValue && rightScale?.(rightValue)) || 0)
+    return yScale(yValue) / numExtraSecondaryBands + rightCoord
+  }
+
+  function getCaseCoords(caseID: string, plotNum = 0) {
+    const xValue = data?.getNumeric(caseID, xAttrID) ?? NaN
+    const yAttrID = yAttrIDs[plotNum]
+    const yValue = data?.getNumeric(caseID, yAttrID) ?? NaN
+    const rightValue = data?.getStrValue(caseID, rightSplitID) ?? ""
+    const rightCoord = (rightValue && rightScale?.(rightValue)) || 0
+    const xCoord = getXCoord(caseID)
+    const yCoord = getYCoord(caseID, plotNum)
+    const color = dataConfiguration?.getLegendColorForCase(caseID)
+    return { xValue, yValue, xCoord, yCoord, rightCoord, color }
+  }
+
+  function residualSquare(slope: number, intercept: number, caseID: string, plotNum = 0): ISquareOfResidual {
+    const { xValue, xCoord, yCoord, rightCoord, color } = getCaseCoords(caseID)
+    const yScale = y2Scale && plotNum === numberOfPlots - 1 ? y2Scale : y1Scale
+    const lineYCoord = yScale(slope * xValue + intercept) / numExtraSecondaryBands + rightCoord
+    const residualCoord = yCoord - lineYCoord
+    const lineXCoord = xCoord + residualCoord
+    const x = Math.min(xCoord, lineXCoord)
+    const y = Math.min(yCoord, lineYCoord)
+    const side = Math.abs(residualCoord)
+    return { caseID, color, side, x, y }
+  }
+
+  function residualSquares(lineDescriptions: ILineDescription[]) {
```

Comment:
I'm having second thoughts about this function name. It's very similar to the `residualSquare` function name and may be confusing in that regard. Would it be better to name the `residualSquare` function something like `residualSquareAttributes` or `residualSquareDescriptions`? Or is there a better name for the `residualSquares` function?

Reply:
Those sound good to me 👍

Reply:
Were you going to rename the second function?